### PR TITLE
net-dns/pihole: prepare /var/lib/pihole/gravity.db file during install

### PR DESCRIPTION
**Pull request by Bogdan107**

### CODE REVIEW

File: net-dns/pihole/pihole-5.3.1-r2.ebuild [hunk 1]
```diff
@@ -11,10 +11,11 @@ SRC_URI="https://github.com/pi-hole/pi-hole/archive/v${PV}.tar.gz -> ${P}.tar.gz
 LICENSE="EUPL-1.2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="cron www"
+IUSE="cron www ftl"
```
Comment (Tatsh) [inline net-dns/pihole/pihole-5.3.1-r2.ebuild:14]: Can we not have the `ftl` in IUSE? It's only used to pull in a package and does not affect configuration/build.

**Review by Tatsh — CHANGES_REQUESTED** [review]

**Comment by Tatsh** [comment]
Otherwise this is good. I already have this in my copy but I haven't applied it to the ebuild.

File: net-dns/pihole/pihole-5.3.1-r2.ebuild (hunk 1 above)
Comment (Bogdan107) [reply net-dns/pihole/pihole-5.3.1-r2.ebuild:14]: Two packages _net-dns/pihole_ and _net-dns/pihole-ftl_ are mutually connected.
Each of them can not be used without another.
So user, which install one of them, in 100% times install another.

If overlay admin want this - I remove this use flag.
But I think, that this use flag MUST be present and activated by default (in form of **+ftl**).

Comment (Tatsh) [reply net-dns/pihole/pihole-5.3.1-r2.ebuild:14]: Remove the IUSE ftl flag, and make `net-dns/pihole-ftl` a required dependency.

Comment (Tatsh) [reply net-dns/pihole/pihole-5.3.1-r2.ebuild]: Actually this change is not required. On line 20, `net-dns/${PN}-ftl` is already there in RDEPEND. So just remove this change.

File: net-dns/pihole/pihole-5.3.1-r2.ebuild [hunk 2]
```diff
@@ -74,6 +79,12 @@ src_install() {
 	einstalldocs
 
 	dobin ${PN}
+	
+	# make sure the working directory exists
+	diropts -m0755
+	keepdir /var/lib/${PN}
+	
+	sqlite3 ${D}/var/lib/${PN}/gravity.db < advanced/Templates/gravity.db.sql
```
Comment (Tatsh) [inline net-dns/pihole/pihole-5.3.1-r2.ebuild:86]: On this line, add to the end: ` || die`.

**Comment by Tatsh** [comment]
Do you know how to rebase all of this into a single commit? If not, I can do it here.

**Comment by Bogdan107** [comment]
> Do you know how to rebase all of this into a single commit? If not, I can do it here.

I done rebase just 2 times of my git experience.
I will training rebase in local computer git tree.
Please, make rebase in public overlay instead of me.

**Pull request merged**